recipes/qwt/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+        qwtconfig_path = os.path.join(self.source_folder, self._source_subfolder, "qwtconfig.pri")
+        qwtconfig = tools.load(qwtconfig_path)
+
+        qwtconfig = "CONFIG += conan_basic_setup\ninclude(../conanbuildinfo.pri)\n" + qwtconfig
+        qwtconfig += "QWT_CONFIG {}= QwtDll\n".format("+" if self.options.shared else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtPlot\n".format("+" if self.options.plot else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtWidgets\n".format("+" if self.options.widgets else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtSvg\n".format("+" if self.options.svg else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtOpenGL\n".format("+" if self.options.opengl else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtMathML\n".format("+" if self.options.mathml else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtDesigner\n".format("+" if self.options.designer else "-")
+        tools.save(qwtconfig_path, qwtconfig)
+
+        # qwtbuild.pri
+        qwtbuild_path = os.path.join(self.source_folder, self._source_subfolder, "qwtbuild.pri")
+        qwtbuild = tools.load(qwtbuild_path)
+        # set build type
+        qwtbuild += "CONFIG -= debug_and_release\n"
+        qwtbuild += "CONFIG -= build_all\n"
+        qwtbuild += "CONFIG -= release\n"
+        qwtbuild += "CONFIG += {}\n".format("debug" if self.settings.build_type == "Debug" else "release")
+        if self.settings.build_type == "RelWithDebInfo":
+            qwtbuild += "CONFIG += force_debug_info\n"
+        tools.save(qwtbuild_path, qwtbuild)
+
+    def build(self):
+        self._patch_qwt_config_files()
+
+        if self.settings.compiler == "Visual Studio":
+            vcvars = tools.vcvars_command(self.settings)
+            self.run("{} && qmake {}".format(vcvars, self._source_subfolder), run_environment=True)
+            self.run("{} && nmake".format(vcvars))
+        else:
+            self.run("qmake {}".format(self._source_subfolder), run_environment=True)
+            self.run("make -j {}".format(tools.cpu_count()))
+
+    def package(self):
+        self.copy("COPYING", src=os.path.join(self._source_subfolder), dst="licenses")
+        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "src"))
+        self.copy("*.dll", dst="bin", keep_path=False)
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.so*", dst="lib", keep_path=False, symlinks=True)
+        self.copy("*.dylib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.includedirs = ['include']
+        if self.settings.build_type == "Debug":
+            if self.settings.os == "Windows":
+                self.cpp_info.libs = ["qwtd"]
+            elif self.settings.os == "Macos":
+                self.cpp_info.libs = ["qwt_debug"]
+            else:
+                self.cpp_info.libs = ["qwt"]
+        else:
+            self.cpp_info.libs = ["qwt"]
```

Comment:
```suggestion
        postfix = ""
        if self.settings.build_type == "Debug":
            if self.settings.os == "Windows":
                postfix += "d"
            elif self.settings.os == "Macos":
                postfix += "_debug"
        self.cpp_info.libs = ["qwt" + postfix]
```

and Macos only or all Apple OS?

Reply:
AFAIU, it's ony for macos: `mac:RET = $$member(LIBRARY_NAME, 0)_debug` on https://sourceforge.net/p/qwt/code/HEAD/tree/tags/qwt-6.1.6/qwtfunctions.pri, mac config is not set by qmake for ios, tvos and watchos

Reply:
@ericLemanissier does it maybe make more sense to patch qwtfunctions.pri to unify suffixes for all platforms ?

Reply:
If we need to change this code, I think the best alternative is to use `tools.collect_libs()` https://docs.conan.io/en/latest/reference/tools.html#tools-collect-libs
I think patches should be used only as last resort.